src/deploy/hosting/convertConfig.ts

```diff
@@ -57,27 +60,99 @@ export async function convertConfig(
     return out;
   }
 
-  const endpointBeingDeployed = (serviceId: string, region: string = "us-central1") => {
+  const endpointFromBackend = (
+    targetBackend: backend.Backend,
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): backend.Endpoint | undefined => {
+    const backendsForId = backend.allEndpoints(targetBackend).filter((endpoint) => {
+      return endpoint.id === serviceId;
+    });
+
+    const matchingBackends = backendsForId.filter((endpoint) => {
+      return (
+        (!region || endpoint.region === region) && (!platform || endpoint.platform === platform)
+      );
+    });
+
+    if (matchingBackends.length > 1) {
+      throw new FirebaseError(`More than one backend found for function name: ${serviceId}.
+      If the function is deployed in multiple regions, you must specify a region.`);
+    }
+
+    if (matchingBackends.length === 1) {
+      const endpoint = matchingBackends[0];
+      if (endpoint && isHttpsTriggered(endpoint)) {
+        return endpoint;
+      }
+    }
+    return undefined;
+  };
+
+  const endpointBeingDeployed = (
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): backend.Endpoint | undefined => {
     for (const { wantBackend } of Object.values(payload.functions || {})) {
-      const endpoint = wantBackend?.endpoints[region]?.[serviceId];
-      if (endpoint && isHttpsTriggered(endpoint) && endpoint.platform === "gcfv2") return endpoint;
+      if (!wantBackend) {
+        continue;
+      }
+      const endpoint = endpointFromBackend(wantBackend, serviceId, platform, region);
+      if (endpoint) {
+        return endpoint;
+      }
     }
     return undefined;
   };
 
-  const matchingEndpoint = async (serviceId: string, region: string = "us-central1") => {
-    const pendingEndpoint = endpointBeingDeployed(serviceId, region);
+  const matchingEndpoint = async (
+    serviceId: string,
+    platform?: string,
+    region?: string
+  ): Promise<backend.Endpoint | undefined> => {
+    const pendingEndpoint = endpointBeingDeployed(serviceId, platform, region);
     if (pendingEndpoint) return pendingEndpoint;
     const backend = await existingBackend(context);
     return allEndpoints(backend).find(
       (it) =>
         isHttpsTriggered(it) &&
-        it.platform === "gcfv2" &&
+        it.platform === platform &&
         it.id === serviceId &&
-        it.region === region
+        (region === undefined || it.region === region)
```

Comment:
Does region specifically need to be undefined? Or would an empty string be a case that needs to be considered?

Normally, you wouldn't make a check against `=== undefined`, preferring instead `!thing` (since undefined is falsey). But if `""` (empty string) is technical valid, and not the same as "not defined", then what is written is correct (and I would put a comment there saying why)

Reply:
I should have made this just ( !region || ...stuff ) like elsewhere. region = "" doesn't make have a reasonable meaning separate from undefined.